sdk/src/Services/DynamoDBv2/Custom/DataModel/ContextInternal.cs

```diff
@@ -475,15 +475,15 @@ private object FromDynamoDBEntry(SimplePropertyStorage propertyStorage, DynamoDB
             if (conversion.HasConverter(targetType))
             {
                 var output = conversion.ConvertFromEntry(targetType, entry);
-                if (flatConfig.RetrieveDateTimeInUtc)
+                if (!flatConfig.RetrieveDateTimeInUtc)
                 {
                     if (targetType == typeof(DateTime))
                     {
-                        return ((DateTime)output).ToUniversalTime();
+                        return ((DateTime)output).ToLocalTime();
```

Comment:
No, this is because the default for `flatConfig.RetrieveDateTimeInUtc` is true instead of false. Because it is true dates will automatically be coming back as UTC. If the flag is set to false we are explicitly converting it to localtime now. So basically the logic is flipped now in v4 on this.

I also added a comment on these in the next commit.